test-framework/common/src/main/java/io/quarkus/test/restassured/QuarkusJackson2ObjectMapperFactory.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.test.restassured;
+
+import java.lang.reflect.Type;
+
+import javax.enterprise.inject.spi.CDI;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.restassured.path.json.mapper.factory.DefaultJackson2ObjectMapperFactory;
+
+/**
+ * A RestAssured Jackson2ObjectMapperFactory that pulls the ObjectMapper from CDI
+ * and delegates to the parent if no CDI bean is found
+ *
+ * It is important that this class not be referenced directly anywhere in the code so the dependencies
+ * on RestAssured and Jackson can remain optional
+ */
+public class QuarkusJackson2ObjectMapperFactory extends DefaultJackson2ObjectMapperFactory {
+
+    @Override
+    public ObjectMapper create(Type type, String s) {
+        ObjectMapper result = CDI.current().select(ObjectMapper.class).get();
```

Comment:
Nice feature. Just curios if this will also work in native IT? https://quarkus.io/guides/getting-started-testing#native-executable-testing

Reply:
What makes you think it won't?

Reply:
I am trying to fully understand the limit of CDI in NativeImageTest and specifically this note `...This supports all the features mentioned in this guide except injecting into tests` from the link above about NativeImageTest.

Reply:
Very good catch. I didn't even think of that TBH

Reply:
Yes, you cannot reuse the same ObjectMapper in native, you will need your own (with the same set of modules).

Reply:
Indeed it won't work. Thanks for pointing it out!

Reply:
Technically we probably could start Arc inside the native test as well and use it to pull out beans, but it's probably more of a hassle than it's worth...